### DISCUSSION TITLE
FIP0001 - FIP not transitioning after "Last Call" ends - new FIP state needed?

### DISCUSSION THREAD

**Discussion by dd45e640b42e6da7da96faee3996ef7c** [filecoin-project/FIPs#677] [Enhancements - Governance]
According to FIP0001 a FIP needs to transition from "Last Call" to "Accepted" / "Final" or "Draft".

> Last Call -- This FIP will be listed prominently on the Filecoin website
> ❌ -- A Last Call which results in material changes or substantial unaddressed technical complaints will cause the FIP to revert to Draft.
> ➡️ Accepted (Core FIPs only) -- A successful Last Call without material changes or unaddressed technical complaints will become Accepted.
> ➡️ Final (Not core FIPs) -- A successful Last Call without material changes or unaddressed technical complaints will become Final.

This creates problems when the governing body, in the case of core FIPs the core devs, did not come to a conclusion to transition the FIP back to "Draft" or forward to "Accepted"/"Final" at the time "Last Call" ends.

Case in point is here FIP0056 that left Last Call without a state transition decided on. This leaves room for unintended actions like calls for extending the "Last Call" state (possibly indefinitely) 

This should be resolved. I propose to add an additional state: 

"Closed", signaling that Last Call has passed but not concluded. 

> Closed -- This FIP las left Last Call without being either transitioned to Draft or Accepted/Final
> ❌ -- Changes to the FIP will cause the FIP to revert to Draft.
> ❌ -- No decision on the FIP transitioning to Accepted/Final will transition the FIP to Rejected. 
> ➡️ Accepted (Core FIPs only) -- A decision on state transition to Accepted will transition the FIP to Accepted.
> ➡️ Final (Not core FIPs) -- A decision on state transition to Final will transition the FIP to Final.

As for "Last Call" i propose a max. 14 day period allowed for this state.

**Comment by kaitlin-beegle** [comment]
This is representative of neither the issue nor the process.  

We move these tags manually. They are not automated; we don't yet have the resources to support automated FIPs labeling. This FIP is currently in Last Call, and we've extended the Last Call period to only end after Core Devs have made a decision.  Once they do, the FIP will be statused as either 'Accepted' or 'Rejected'. 

In addition, because we change tags manually, we usually wait until the following day just to be inclusive of all time zones.  Regardless, implementations happen weeks or months later, so there is not any immediate impact on the status of the network and/or FIP incorporation. 

'Closed' does not reflect a use case that is common or necessary.

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
the use case:

- the governing body that has to make a decision doesn't make a decision after Last Call ends (eg. FIP0056, with  an original  deadline set at 24th)

what if the core devs ask for more time to review the input of last call after the extension? how long are we keeping this in Last Call before someone makes a decision?

this is in no way FIP0056 specific but a general problem in the FIP process.

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
the above is just formalizing what you describe. why we have a standardized FIP process if we do not follow it?

 > we've extended the Last Call period to only end after Core Devs have made a decision
 
 **a unilateral decision (same is valid for last call deadlines tbh) is not how this should work.** 

we do not need to set deadlines to Last Calls if the boundary is the decision and not the deadline!

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
the above is a try to avoid having FIPs in a mutable state forever. it is currently not clear when a FIP becomes immutable.

if FIP0056 would be considered immutable with entering Last Call then the extension wouldn't be needed and we can just use the decision as the deadline to get it implemented. this all is very murky in the end. it should be formalized

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
https://filecoinproject.slack.com/archives/C01EU76LPCJ/p1679702482675479
 
 as a reference.

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
@kaitlin-beegle can you point me to the place where the whole Last Call handling procedure (setting deadlines etc.) is documented right now?
